### ISSUE TITLE
Add params to Booster object for feature contribution computing

### ISSUE THREAD

**Issue by yanyachen** [Microsoft/LightGBM#288]
To implement the idea described at http://blog.datadive.net/interpreting-random-forests/, I think it's necessary to add params to Booster object (public or private?) so that we can scale down the raw prediction (e.g. log odds) of each node/leaf to get the real prediction (e.g. probability).

Does this sound a reasonable change to you @guolinke ? If so, where do you prefer to add the params to Booster? Thanks you!

**Comment by yanyachen** [comment]
Sorry, another question.
Is there anyway to get the learning_rate of each tree? The learning rate may change along training process if using reset_parameter.

**Comment by wxchan** [comment]
to get the learning_rate of each tree:

![figure_1](https://cloud.githubusercontent.com/assets/13624043/22776685/94557902-eeeb-11e6-9b58-250ddbf371bf.png)

the learning rate is multiplied to leaf value, so `learning_rate = (left leaf value * left leaf count + right leaf value * right leaf count)/(interval value * interval count)`

for figure above, learning_rate = (0.058523409363745504*1666 + 0.08783041641520822*3314) / (4980*0.6002008032128514) = 0.13

**Comment by guolinke** [comment]
@yanyachen 
I think the solution provided by @wxchan can solve your problem without add additional parameters.

**Comment by yanyachen** [comment]
Yes, it solved my problem. Sorry, I forgot to close the issue.

**Comment by yanyachen** [comment]
I just realized that if the depth of tree is only 1, then this method can't be applied because interval_value is always 0. Is there anyway to walk around this? Thanks you!

**Comment by guolinke** [comment]
@yanyachen I will open a new PR to output shrinkage_rate for each tree
